examples/go/tagging/handler/handler.go

```diff
@@ -31,140 +32,145 @@ type vcConfig struct {
 }
 
 // The structure for a Cloud Event.
-type vmEvent struct {
-	Event event `json:"data"`
-}
-type event struct {
-	Vm vmEventArgument `json:"Vm"`
-}
-type vmEventArgument struct {
-	Vm managedObjectReference `json:"Vm"`
-}
-type managedObjectReference struct {
-	Type  string `json:"Type"`
-	Value string `json:"Value"`
+type incoming struct {
+	Data types.Event `json:"data,omitempty"`
 }
 
+var (
+	client *vsClient.Client
+	ctx    = context.Background()
+	cfg    vcConfig
+	sigCh  = make(chan os.Signal)
+)
+
 // Handle a function invocation
 func Handle(req handler.Request) (handler.Response, error) {
-	ctx := context.Background()
-
-	// Retrieve vCenter and tag info from the config file.
-	cfg, err := loadTomlCfg()
-	if err != nil {
-		log.Error(err)
-		return
+	// Connect to vSphere govmomi API if not already done. Also, load the required
+	// vcconfig.
+	if client == nil {
+		err := vsConnect()
+		if err != nil {
+			log.Fatalf("connect to vSphere failed: %v", err)
+			return handler.Response{
+				Body:       nil,
+				StatusCode: http.StatusBadRequest,
+			}, fmt.Errorf("connect to vSphere failed: %w", err)
+		}
 	}
 
-	// Incoming request object.
-	var event vmEvent
-	err := json.Unmarshal(req.Body, &event)
+	// Set up os signal handling to log out of vSphere.
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+	go handleSignal(ctx, sigCh)
+
+	// Retrieve the Managed Object Reference from the event.
+	moRef, err := parseEventMoRef(req.Body)
 	if err != nil {
+		log.Fatalf("retrieve managed reference object failed: %v", err)
 		return handler.Response{
 			Body:       nil,
 			StatusCode: http.StatusBadRequest,
 		}, err
 	}
 
-	if event.Event.Vm.Vm.Value == "" || event.Event.Vm.Vm.Type == "" {
+	log.Println("tag VM moRef with tagID (URN)")
```

Comment:
Some (if not all) of these `log.Println` statements can be considered DEBUG information. You might want to only log them if an appropriate ENV var is set. Reason: if your fn gets popular and has a lot of invocations, it could flood the container host without providing any real value.

Reply:
I'm not sure how to pass an environment variable to `handler.go`. Do you think it would be all right to add a section to vcconfig.toml, maybe something like the following?

```toml
[debug]
verbose = true
```

And then, I can do something like, 

```go
if verbose == true {
    log.Println("the error and success messages")
}
```